### ISSUE TITLE
Remove extension mock modules

### ISSUE THREAD

**Issue by shawkins** [fabric8io/kubernetes-client#3924]
### Is your task related to a problem? Please describe

After #3912 the extension mock modules are no longer needed.

### Describe the solution you'd like

The user facing classes will have been deprecated, so this issue is a reminder to remove them at a later time - 6.1?

### Additional context

The usage of these modules are no longer required since the generic Kubernetes Mock Server can be used instead (is preferred).

> - Extension specific `EnableXXXMockClient` and `XXXMockServer` classes have been deprecated.  You can simply use `EnableKubernetesMockClient` and `KubernetesMockServer` instead. Dependencies on the xxx-mock jar are then no longer needed, just a dependency to `kubernetes-server-mock`.

https://github.com/fabric8io/kubernetes-client/blob/43af167c663031dac37f08dda36e35e512462071/doc/MIGRATION-v6.md#L238

**Comment by anurag-rajawat** [comment]
@shawkins Do I need to remove the `<module>mock</module>` line from the `pom.xml` files present in extensions sub-modules or completely remove the mock modules present in the extensions sub-modules?

**Comment by shawkins** [comment]
@anurag-rajawat they should not be removed yet.  To ease the migration to 6.0 they will still initially be present but deprecated.  In a subsequent release they will be removed.

**Comment by shawkins** [comment]
@anurag-rajawat or any other community user interested in picking up an issue, we are ready for this.  It is mostly removing the modules, but it will also require cleaning up any tests and examples that still reference them.

**Comment by anurag-rajawat** [comment]
@shawkins I would love to work on this, assign this to me.
Thanks!
